### PR TITLE
Damping mask

### PR DESCRIPTION
**Pull request by sghelichkhani**
This PR introduces a vertical masking function to  allow damping only within thermal boundary layers.

### CODE REVIEW

File: demos/adjoint/inverse.py [hunk 1]
```diff
@@ -35,6 +47,7 @@ def inverse(alpha_u, alpha_d, alpha_s):
         mesh = f.load_mesh("firedrake_default_extruded")
 
     bottom_id, top_id, left_id, right_id = "bottom", "top", 1, 2
+    X = SpatialCoordinate(mesh)  # spatial coordinate
```
Comment (angus-g) [inline demos/adjoint/inverse.py:50]:
```suggestion
    X = SpatialCoordinate(mesh)
```

File: demos/adjoint/inverse.py [hunk 2]
```diff
@@ -15,6 +15,18 @@ def main():
     inverse(alpha_u=1e-1, alpha_d=1e-2, alpha_s=1e-1)
 
 
+def gaussian(r, r_0, sigma):
+    """
+    Returning a gaussing distribution around r_0 with std deviation of sigma
```
Comment (angus-g) [inline demos/adjoint/inverse.py:20]:
```suggestion
    Return a Gaussian distribution centred around r_0 with standard deviation of sigma
```

**Review by angus-g — CHANGES_REQUESTED** [review]
Seems sensible, just a few typos to fix.

**Review by angus-g — APPROVED** [review]
All good now, I think! Wait for tests to go green and squash it all in!

**Comment by sghelichkhani** [comment]
Lovely! Thanks Angus!

**Pull request merged**